src/LiveComponent/src/Metadata/LivePropMetadata.php

```diff
@@ -135,7 +135,7 @@ public function withModifier(object $component): self
             throw new \LogicException(\sprintf('Method "%s::%s()" given in LiveProp "modifier" does not exist.', $component::class, $modifier));
         }
 
-        $modifiedLiveProp = $component->{$modifier}($this->liveProp);
+        $modifiedLiveProp = $component->{$modifier}($this->liveProp, $this->getName());
```

Comment:
For a change like this, a test would be needed. Especially for things related to query, URL, etc.

(The fact tests do pass after your changes does not proves it does work (nor that our current suite of test is good enough, to be fair))

Reply:
Added to the existing test, I don't see the need to make a separate test for just this param unless there are some edge cases that aren't entirely obvious to me at this time.

Reply:
Let's test a page with two instances of same component, different prefix used in the modifier ... and check that the correct one as the good value and not the other :)